src/background/utils/script.js

```diff
@@ -121,3 +121,60 @@ export function getNameURI(script) {
   if (!ns && !name) nameURI += script.props.id || '';
   return nameURI;
 }
+
+/**
+ * @param {VMScript} script
+ * @returns {string | undefined}
+ */
+export function inferScriptHome(script) {
+  let u = script.custom.lastInstallURL;
+  if (u) {
+    u = u.split('/', 6);
+    switch (u[2]) {
+    case 'greasyfork.org':
+      if (u[3] !== 'scripts') u.splice(3, 1);
+      break;
+    case 'raw.githubusercontent.com':
+      u[2] = 'github.com';
+      break;
+    case 'github.com':
+      break;
+    case 'openuserjs.org':
+      u[3] = 'scripts';
+      u[4] = u[4].replace(/(\.min)?\.user\.js$/, '');
+      break;
+    default:
+      u = false;
+    }
+    if (u) {
+      u.length = 5; // scheme + 1 + host + group + name
+      u = u.join('/');
+    }
+  }
+  if (!u) {
+    u = script.meta.namespace;
+    u = /^https?:\/\/(?!tampermonkey\.net\/)/.test(u)
+      && getFullUrl(u).replace(/^https?(:\/\/userscripts)(\.org\/users\/\w)/, 'https$1-mirror$2');
+  }
+  if (u) {
+    script.props[HOMEPAGE_URL] = u;
```

Comment:
Can we infer it on demand during the render? I guess we don't need to store the inferred URLs.

Reply:
I thought it'd be wasteful to run those regexps on all scripts every time. I guess I can calculate them in GetData when opening the popup, it'll be rare enough.